lib/bom.js

```diff
@@ -362,6 +366,55 @@ declare class MutationObserver {
     disconnect(): void;
 }
 
+declare class DOMRectReadOnly {
+    x: number,
+    y: number,
+    width: number,
+    height: number,
+    top: number,
+    right: number,
+    bottom: number,
+    left: number,
+    constructor(x: number, y: number, width: number, height: number): void,
+    fromRect(rectangle?: { x?: number, y?: number, width?: number, height?: number }): DOMRect,
+}
+
+declare class DOMRect extends DOMRectReadOnly {
+    constructor(x: number, y: number, width: number, height: number): void,
+}
+
+declare type IntersectionObserverEntry = {
+    boundingClientRect: DOMRectReadOnly,
+    intersectionRatio: number,
+    intersectionRect: DOMRectReadOnly,
+    isIntersecting: boolean,
+    rootBounds: DOMRectReadOnly,
+    target: HTMLElement,
+    time: DOMHighResTimeStamp,
+};
+
+declare type IntersectionObserverCallback = (
+    entries: Array<IntersectionObserverEntry>,
+    observer: IntersectionObserver,
+) => any;
+
+declare type IntersectionObserverOptions = {
+    root?: Node | null,
+    rootMargin?: string,
+    threshold?: number | Array<number>,
+};
+
+declare class IntersectionObserver {
+    constructor(
+      callback: IntersectionObserverCallback,
+      options: IntersectionObserverOptions
```

Comment:
The second argument is actually optional, see https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/IntersectionObserver

Reply:
There is an open PR fixing this: https://github.com/facebook/flow/pull/4664